EIPS/eip-2602.md

```diff
@@ -0,0 +1,50 @@
+---
+eip: 2602
+title: Disable null hash message verification for ecrecover precompile
+author: Wei Tang (@sorpaas)
+discussions-to: https://github.com/sorpaas/EIPs/issues/10
+status: Draft
+type: Standards Track
+category: Core 
+created: 2020-04-15
+---
+
+## Simple summary
+
+Disable null hash message `ecrecover` operation in the precompile, due to it being potentially unsafe.
+
+## Abstract
+
+In `ecrecover`, disable null hash message support by acting the same as the operation fails.
+
+## Motivation
+
+In secp256k1, null hash message verification is unsafe because an adversary can [forge the signature](https://crypto.stackexchange.com/questions/50279/how-should-ecdsa-handle-the-null-hash/50290#50290). Many libraries also by default disallow verification of null hash message. By disabling null hash message verification, we avoid potential misuse of the precompile.
+
+## Specification
+
+In the beginning of execution of `ecrecover` precompile, fetch the message and check if it is equal to null hash (`0x0000...0000`). If so, set output length to `0` and return.
```

Comment:
Can this specification a bit more extended? Should clarify if "message" here means a part of the input to the precompile or the entire input?

Also probably spelling out the entire hash is fine as that would signal how many bits it is (256?).